src/pages/o/[orgId]/surveys/[surveyId]/index.tsx

```diff
@@ -1,32 +1,346 @@
+      `/api/orgs/${orgId}/surveys/${surveyId}`
+    );
+  } catch (e) {
+    return { notFound: true };
+  }
+
+  if (req.method === 'POST') {
+    const form = await parseRequest(req);
+    const responses: Record<
+      string,
+      {
+        options?: number[];
+        question_id: number;
+        response?: string;
+      }
+    > = {};
+
+    for (const name in form) {
+      const isSignature = name.startsWith('sig');
+      const isPrivacy = name.startsWith('privacy');
+      const isMetadata = isSignature || isPrivacy;
+      if (isMetadata) {
+        continue;
+      }
+
+      const fields = name.split('.');
+      const questionId = fields[0];
+      const questionType = fields[1];
+
+      if (typeof responses[questionId] === 'undefined') {
+        responses[questionId] = {
+          question_id: parseInt(fields[0]),
+        };
+      }
+
+      if (questionType == 'options') {
+        if (Array.isArray(form[name])) {
+          responses[questionId].options = (form[name] as string[]).map((o) =>
+            parseInt(o)
+          );
+        } else {
+          responses[questionId].options = [parseInt((form[name] as string)!)];
+        }
+      } else if (questionType == 'text') {
+        responses[questionId].response = form[name] as string;
+      }
+    }
+
+    let signature: null | {
+      email: string;
+      first_name: string;
+      last_name: string;
+    } = null;
+    // TODO: handle other signature types
+    if (form.sig == 'email') {
+      signature = {
+        email: form['sig.email'] as string,
+        first_name: form['sig.first_name'] as string,
+        last_name: form['sig.last_name'] as string,
+      };
+    }
+
+    try {
+      await apiClient.post(
+        `/api/orgs/${orgId}/surveys/${surveyId}/submissions`,
+        {
+          responses: Object.values(responses),
+          signature,
+        }
+      );
+    } catch (e) {
+      status = 'error';
+    }
+    if (status !== 'error') {
+      status = 'submitted';
+    }
+
+    if (status === 'submitted') {
+      res.writeHead(302, {
+        Location: `/o/${orgId}/surveys/${surveyId}/submitted`,
+      });
+      res.end();
+    }
+  }
 
   return {
     props: {
       orgId,
-      surveyId,
+      status,
+      survey,
     },
   };
 }, scaffoldOptions);
 
 type PageProps = {
   orgId: string;
-  surveyId: string;
+  status: FormStatus;
+  survey: ZetkinSurveyExtended;
 };
 
-const Page: FC<PageProps> = ({ orgId, surveyId }) => {
+type FormStatus = 'editing' | 'invalid' | 'error' | 'submitted';
+type SignatureOption = 'authenticated' | 'email' | 'anonymous';
+
+function RadioFormControlLabel(props: FormControlLabelProps) {
+  const radioGroup = useRadioGroup();
+
+  let checked = false;
+
+  if (radioGroup) {
+    checked = radioGroup.value === props.value;
+  }
+
+  if (checked) {
+    return (
+      <FormControlLabel
+        checked={checked}
+        sx={{
+          backgroundColor: '#fbcbd8',
```

Comment:
Ideally, we'd use a color from the theme, that can be retrieved using `useTheme()`.

Reply:
Discussion continues over at https://github.com/zetkin/app.zetkin.org/pull/1665#discussion_r1405212639.